### ISSUE TITLE
Tensorflow support

### ISSUE THREAD

**Issue by bratyslav** [ZHKKKe/MODNet#66]
Thank you for your job!
Is there an easy way to convert your PyTorch model to Tensorflow?

**Comment by manthan3C273** [comment]
Hopefully, you can convert PyTorch model to ONNX first. And then convert ONNX to Tensorflow. But currently, it mostly supports popular models. In onder to convert, MODNet into Tensorflow, we've ONNX right now, but some features does not support while converting it to Tensorflow. Possilbe solution might be a minor tweaking to the original model architecture. Maybe this will help.

**Comment by bratyslav** [comment]
Yes, I have some troubles to convert your ONNX model to Tensorflow. Where can I find the original model architecture? Is there a special commit?
Thanks!

**Comment by ZHKKKe** [comment]
@bratyslav 
Hi, thanks for your attention.
I am sorry that we do not have any plan of the Tensorflow version right now.
Hope the community can help. :)

**Comment by bratyslav** [comment]
> Hopefully, you can convert PyTorch model to ONNX first. And then convert ONNX to Tensorflow. But currently, it mostly supports popular models. In onder to convert, MODNet into Tensorflow, we've ONNX right now, but some features does not support while converting it to Tensorflow. Possilbe solution might be a minor tweaking to the original model architecture. Maybe this will help.

Have you tried to convert onnx to coreml? I tried to use this code:

`import coremltools

mlmodel = coremltools.converters.onnx.convert(
    model = 'pretrained/modnet.onnx',
    predicted_feature_name = [],
    minimum_ios_deployment_target='13',    
)`

and get error: 'index -2 is out of bounds for axis 0 with size 0'

**Comment by ZHKKKe** [comment]
@bratyslav 
Sorry, I haven't tried it. Hope someone can help.

**Comment by xpeng** [comment]
> > Hopefully, you can convert PyTorch model to ONNX first. And then convert ONNX to Tensorflow. But currently, it mostly supports popular models. In onder to convert, MODNet into Tensorflow, we've ONNX right now, but some features does not support while converting it to Tensorflow. Possilbe solution might be a minor tweaking to the original model architecture. Maybe this will help.
> 
> Have you tried to convert onnx to coreml? I tried to use this code:
> 
> ```
> 
> mlmodel = coremltools.converters.onnx.convert(
>     model = 'pretrained/modnet.onnx',
>     predicted_feature_name = [],
>     minimum_ios_deployment_target='13',    
> )
> ```
> 
> and get error: 'index -2 is out of bounds for axis 0 with size 0'

same issue with you. 
and i try convert ckpt to onnx using export_onnx under different torch version, like through `pip install "torch==1.4" "torchvision==0.5.0"`, there comes different error, such as "while converting op of type: Upsample..."

**Comment by ZHKKKe** [comment]
@xpeng 
As said in the [README.md](https://github.com/ZHKKKe/MODNet/tree/master/onnx) file, the tool requires: 
```
onnx==1.8.1
onnxruntime==1.6.0
opencv-python==4.5.1.48
torch==1.7.1
```
Since this tool is provided by the community, we have not verified its compatibility with other PyTorch versions. Sorry for this.
